openmdao/solvers/nonlinear/nonlinear_block_gs.py

```diff
@@ -101,6 +101,9 @@ def _iter_initialize(self):
         if system.under_complex_step and self.options['cs_reconverge']:
             system._outputs._data += np.linalg.norm(system._outputs._data) * 1e-10
 
+        # Execute guess_nonlinear if specified.
+        system._guess_nonlinear()
```

Comment:
Should we check system._has_guess here before calling _guess_nonlinear?  I noticed that ImplicitComponent does some stuff in its _guess_nonlinear that isn't needed if no guesses are provided.  Broyden does the same thing.  Pretty minor, but still...

Reply:
Well, we don't do it in Newton. The only place we check it is in Group._guess_nonlinear.  I have added it into ImplicitComponent._guess_nonlinear to be consistent.

Reply:
Note though, the only time where this will come into play is the case where a nonlinear solver is plugged into a component.  When the solver is in a group, we never recurse down to the component level if there isn't a guess.